app-modules/knowledge-base/src/Filament/Actions/DraftKnowledgeBaseArticleWithAiAction.php

```diff
@@ -0,0 +1,151 @@
+                    ->required(),
+            ])
+            ->action(function (array $data, Get $get, Set $set, Page $livewire) {
+                $service = Feature::active('ai-integrated-assistant-settings')
+                    ? app(AiIntegratedAssistantSettings::class)->default_model->getService()
+                    : app(AiSettings::class)->default_model->getService();
+
+                $userName = auth()->user()->name;
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $articleTitle = $livewire?->data['title'];
+
+                try {
+                    $content = $service->complete(
+                        <<<EOL
+                            My name is {$userName}, and I am a {$userJobTitle} at {$clientName}. I am currently editing an article in the knowledge base of the college. The current title is "{$articleTitle}". 
+
+                            You should only respond with the article content, you should never greet them.
+                            The first line should contain the raw title of the article, with no "Title: " label at the start.
+                            All following lines after the title are the article body.
+
+                            When you answer, it is crucial that you format the article body using rich text in Markdown format.
+                            The title line can not use Markdown formatting, it is plain text.
+
+                            Then, write the article content on the following instructions provided by user.
+
+                            Here are the details:
+                        EOL,
+                        $data['instructions']
+                    );
+                } catch (MessageResponseException $exception) {
+                    report($exception);
+
+                    Notification::make()
+                        ->title('AI Assistant Error')
+                        ->body('There was an issue using the AI assistant. Please try again later.')
+                        ->danger()
+                        ->send();
+
+                    $this->halt();
+
+                    return;
+                }
+
+                $set('title', (string) str($content)
+                    ->before("\n")
+                    ->trim());
+
+                $set('article_details', (string) str($content)->after("\n")->markdown());
+            })
+            ->visible(
+                auth()->user()->hasLicense(LicenseType::ConversationalAi)
+            );
+    }
+
+    public static function getDefaultName(): ?string
+    {
+        return 'draftWithAi';
+    }
+
+    public function mergeTags(array | Closure $tags): static
+    {
+        $this->mergeTags = $tags;
+
+        return $this;
+    }
+
+    public function getMergeTags(): array
+    {
+        return $this->evaluate($this->mergeTags);
+    }
```

Comment:
Please remove the merge tag methods and the merge tags property from here as they aren't used in this context